src/main/java/codeu/controller/GIFSearchServlet.java

```diff
@@ -0,0 +1,114 @@
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.json.JSONObject;
+import org.json.JSONArray;
+
+public class GIFSearchServlet extends HttpServlet {
+
+	private final String USER_AGENT = "Mozilla/5.0";
+	
+	@Override
+	public void doGet(HttpServletRequest request, HttpServletResponse response)
+	        throws IOException, ServletException {
+		String GIFsearch = java.net.URLEncoder.encode(request.getParameter("search"), "UTF-8");
+		GIFsearch = clean(GIFsearch);
+		
+		String htmlResponse = "";
+		try {
+			htmlResponse = sendGet(GIFsearch);
+		} catch (Throwable e) {
+			for(int i = 0; i < 20; i++) {
+				htmlResponse = "<div class='row'>"
+						+ "<div class='col'><div data-dismiss='modal'><img onclick='sendGIF(this)' width=200 src='https://i.giphy.com/bi6RQ5x3tqoSI.gif' class='img-fluid' alt='Responsive image'></div></div>"
+						+ "<div class='col'><div data-dismiss='modal'><img onclick='sendGIF(this)' width=200 src='https://i.giphy.com/bi6RQ5x3tqoSI.gif' class='img-fluid' alt='Responsive image'></div></div>"
+						+ "</row>";
+        
+			}
+
+        }
+		
+		response.setContentType("text/plain");
+	    response.setCharacterEncoding("UTF-8");
+	    response.getWriter().write(htmlResponse);
+	    
+	}
+	
+	// HTTP GET request
+	private String sendGet(String GIFsearch) throws Exception {
+
+		String url = "http://api.giphy.com/v1/gifs/search?q="+GIFsearch+"&api_key=iDKR9vNZ9WlR2B1aqbmtentOM2wUjLyZ&limit=20";
+		
+		URL obj = new URL(url);
+		HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+
+		// optional default is GET
+		con.setRequestMethod("GET");
+
+		//add request header
+		con.setRequestProperty("User-Agent", USER_AGENT);
+
+		int responseCode = con.getResponseCode();
+		//System.out.println("\nSending 'GET' request to URL : " + url);
+		//System.out.println("Response Code : " + responseCode);
+
+		BufferedReader in = new BufferedReader(
+		        new InputStreamReader(con.getInputStream()));
+		String inputLine;
+		StringBuffer response = new StringBuffer();
+
+		while ((inputLine = in.readLine()) != null) {
+			response.append(inputLine);
+		}
+		in.close();
+		
+		
+        //Read JSON response and add URLs to ArrayList
+        JSONObject giphyJSONResponse = new JSONObject(response.toString());
+        JSONArray JSONArray = giphyJSONResponse.getJSONArray("data");
+        
+	    String htmlResponse = "";
+	    
+        for(int i=0; i < JSONArray.length(); i++) {
+            JSONObject JSONObject = JSONArray.getJSONObject(i);
+            String id = JSONObject.getString("id");
+            String gifURL = "https://i.giphy.com/"+id+".gif";
+            
+            if (i%2 == 0) {
+            	if (i!=0) {
+            		htmlResponse+="</div>";
+            	}
+            	htmlResponse+="<div class='row'>";
+    	    }
+            htmlResponse+="<div class='col'><div data-dismiss='modal'>"
```

Comment:
Hm, it might be cleaner to build this result with a JSP file instead of inline like this. That can be a separate cleanup PR, if you'd like to look into it!